### PR TITLE
feat: granular git status

### LINKED ISSUE #31644: POC for granular git status - Sunlife

### PR DESCRIPTION
**Pull request by brayn003**
## Description
Lists down changes in entity with level 1 granularity

Fixes #31644 

## Automation

/ok-to-test tags="@tag.Git"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/8642841391>
> Commit: dd9dd7abb619a4e62edd950cbc6e64e15b4a42e2
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=8642841391&attempt=1" target="_blank">Click here!</a>

## Summary by CodeRabbit

- **New Features**
	- Introduced granular tracking of Git changes, including added, modified, and removed pages, datasources, queries, JS objects, and JS libraries.
	- Implemented a new feature flag `release_git_status_granular_enabled` to toggle granular Git status tracking.
	- Added new components to display detailed Git changes with expandable views for different types of entities.

- **Enhancements**
	- Simplified the logic for determining if a commit is required by enhancing the Git status checks.
	- Enhanced Git status display to include specific messages and icons for various types of changes.
	- Improved the Git sync feature's user interface with updated components and styling for displaying Git changes.
	
- **Bug Fixes**
	- Corrected the display logic to accurately reflect when a JS library is added instead of modified.

- **Refactor**
	- Refactored Git status handling logic on both client and server sides for improved performance and readability.
	- Updated test suites to align with the new Git status tracking and display functionalities.

- **Tests**
	- Added and updated tests to cover new Git status functionalities and components.

### CODE REVIEW

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/dtos/ce/GitStatusCE_DTO.java [hunk 1]
```diff
@@ -11,47 +11,98 @@
...
-    // Name of added resources to local git repo
+    // set of files that were added anew
     Set<String> added;
 
-    // Name of deleted resources from local git repo
+    // set of files that were removed
     Set<String> removed;
 
-    // Name of conflicting resources
-    Set<String> conflicting;
```
Comment (nidhi-nair) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/dtos/ce/GitStatusCE_DTO.java:24]: QQ, was this property not being used earlier?

**Review by nidhi-nair — APPROVED** [review]
LGTM, minor question. Thanks for adding granular tests for each resource!

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/dtos/ce/GitStatusCE_DTO.java (hunk 1 above)
Comment (brayn003) [reply app/server/appsmith-interfaces/src/main/java/com/appsmith/external/dtos/ce/GitStatusCE_DTO.java:24]: it is, I think git messed up the order, it's present in both

**Review by ankitakinger — APPROVED** [review]
FE changes look good to me

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/dtos/ce/GitStatusCE_DTO.java (hunk 1 above)
Comment (nidhi-nair) [reply app/server/appsmith-interfaces/src/main/java/com/appsmith/external/dtos/ce/GitStatusCE_DTO.java:24]: Oh yeah, my bad! I see it now. :)

**Pull request merged**